common/chan_utils_test.go

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2024 Function Stream Org.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestSendToChannel(t *testing.T) {
+
+	//send data
+	//The first t.Run is a buffered chan, and the second t.Run is an unbuffered chan
+	t.Run("send success,Buffered chan", func(t *testing.T) {
+		c := make(chan string, 1)
+		ctx := context.Background()
+		if !SendToChannel(ctx, c, "data") {
+			t.Fatal("SendToChannel should return true when sending succeeds")
+		}
+		value := <-c
+		// Verify if the received data is correct
+		if value != "data" {
+			t.Errorf("expected to receive \"data\" from channel, but received %s", value)
+		}
+	})
+
+	t.Run("send success,Unbuffered chan", func(t *testing.T) {
+		c := make(chan string)
+		ctx := context.Background()
+		var wg sync.WaitGroup
+		wg.Add(2)
+		go func() {
+			defer wg.Done()
+			SendToChannel(ctx, c, "data")
+		}()
+		go func() {
+			defer wg.Done()
+			value := <-c
+			// Verify if the received data is correct
+			if value != "data" {
+				t.Errorf("expected to receive \"data\" from channel, but received %s", value)
+			}
+		}()
```

Comment:
You just need to use another goroutine to send the data and receive & check the data in the testing main goroutine.